src/pystack/_pystack/mem.cpp

```diff
@@ -155,36 +158,156 @@ MemoryMapInformation::setHeap(const VirtualMap& heap)
     d_heap = heap;
 }
 
+template <typename K, typename V>
+LRUCache<K,V>::LRUCache(size_t capacity)
+    : cache_capacity(capacity)
+    , size(0) {};
+
+template <typename K, typename V>
+void
+LRUCache<K,V>::put(const K& key, const V&& value, size_t value_size)
+{
+    if(value_size > cache_capacity){
+        return;
+    }
+
+    auto it = cache.find(key);
+
+    if (it != cache.end()) {
+        cache_list.erase(it->second.second);
+        cache.erase(it);
+    }
+
+    while (size + value_size > cache_capacity) {
+        auto last = --cache_list.end();
```

Comment:
It would be a little bit cleaner to use:
```suggestion
        auto last = cache_list.rbegin();
```
That gives you a reverse iterator to the first element in reverse order, so the last element in the list.